### ISSUE TITLE
[TF2DS] [WSL2 + Ubuntu] srcds_linux unexpectedly kills itself when game files are under mounted Windows filesystem

### ISSUE THREAD

**Issue by kumfc** [ValveSoftware/Source-1-Games#5419]
This is a very unusual setup to begin with, but I really liked the idea of having a shared folder for both Windows and Linux srcds (because I have not so much space on the system disk where WSL2 filesystem lies). The issue, however, might be generally tied to the filesystem type and cause the same problem on regular Linux hosts as well.

**Reproduction steps:**
1) Download latest TF2DS for Linux via SteamCMD and put it in the mounted Windows filesytem (e.g. /mnt/d/Steam/steamapps/...)
2) Launch it (sv_pure cvar value doesn't matter):
`./srcds_run -console -game tf +sv_pure 0 +map ctf_2fort +maxplayers 24`
3) Server will start loading resources and suddenly crash with the custom OOM error (while taking less than 100m of memory):
`Out of memory or address space. Texture quality setting may be too high.`

**srcds_linux strace before crash:**
```
futex(0x853d5e8, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, {tv_sec=1705537963, tv_nsec=317066000}, FUTEX_BITSET_MATCH_ANY) = -1 ETIMEDOUT (Connection timed out)
futex(0x853d5a8, FUTEX_WAKE_PRIVATE, 1) = 0
sched_setscheduler(79, SCHED_OTHER, [0]) = 0
openat(AT_FDCWD, "/proc/self/status", O_RDONLY) = 4
read(4, "Name:\tsrcds_linux\nUmask:\t0022\nSt"..., 1023) = 1023
close(4)                                = 0
futex(0x853d5e8, FUTEX_WAIT_BITSET_PRIVATE|FUTEX_CLOCK_REALTIME, 0, {tv_sec=1705537993, tv_nsec=318380000}, FUTEX_BITSET_MATCH_ANY) = 0
futex(0x853d5a8, FUTEX_WAKE_PRIVATE, 1) = 0
futex(0x853d4c0, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x853d480, FUTEX_WAKE_PRIVATE, 1) = 1
openat(AT_FDCWD, "/mnt/d/Steam/steamapps/common/TF2_srcds/gameinfo.txt", O_RDONLY) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/mnt/d/Steam/steamapps/common/TF2_srcds", O_RDONLY|O_NONBLOCK|O_LARGEFILE|O_CLOEXEC|O_DIRECTORY) = 4
statx(4, "", AT_STATX_SYNC_AS_STAT|AT_NO_AUTOMOUNT|AT_EMPTY_PATH, STATX_BASIC_STATS, {stx_mask=STATX_BASIC_STATS|STATX_MNT_ID, stx_attributes=0, stx_mode=S_IFDIR|0777, stx_size=4096, ...}) = 0
brk(0x8702000)                          = 0x8702000
getdents64(4, 0x86d92ec /* 12 entries */, 32768) = 368
close(4)                                = 0
openat(AT_FDCWD, "/mnt/d/Steam/steamapps/common/TF2_srcds/tf/gameinfo.txt", O_RDONLY) = 4
stat64("/mnt/d/Steam/steamapps/common/TF2_srcds/tf/gameinfo.txt", {st_mode=S_IFREG|0777, st_size=3706, ...}) = 0
write(3, "Out of memory or address space. "..., 74Out of memory or address space.  Texture quality setting may be too high.
) = 74
exit_group(1)                           = ?
+++ exited with 1 +++
```

The crash seem to be artificial because with sv_pure 1 server successfully loads the VPK content before crashing. 

------------------------

If you're encountered the same problem with WSL, the only way to run the server is to copy/move it to the Linux filesystem where it works as expected.

**Comment by worMatty** [comment]
I dabbled with a TF2 server running under WSL2 some time ago. The server was installed on the Linux drive, and my custom maps were on the mounted Windows drive. It was very slow to load maps. I searched the web for answers and found that WSL2 is reportedly particularly slow at reading files on mounted Windows drives. The proposed solution was to use WSL1. I don't know if this has improved since then.

**Comment by kumfc** [comment]
I tried using WSL1 first, but it does support only x64 binaries. With QEMU and binfmt the srcds_linux starts but freezes at some point. It's probably still possible to trick it around but I decided switching to WSL2 instead.

This custom OOM message seem to be a general response for dozen of problems on OSX and Linux. While googling I've seen several unresolved thrads with it. So I guess it might be not just a WSL mount handling problem but some general NTFS confusion on Linux (maybe).
